Windows build with Clang

danielnachun opened the pull request.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

danielnachun commented:
I was able to make a lot of progress in this iteration of the Windows build, though we're still not there just yet. 

First, I was able to isolate the build failure to this single line:
```
static constexpr T value = std::is_floating_point<T>::value ? NAN : std::numeric_limits<T>::min();
```
the build finishes meaning that all of the other instances of `constexpr` aren't a problem for this Clang + MSVC C++ toolchain.

However, changing `constexpr` to `const` emits a warning from `-Wstatic-float-init`, which by default causes a failure.  Setting `-Wno-static-float-init` with `PKG_CXXFLAGS` allows the build finish successfully.

Unfortunately, the failure point now is much harder to diagnose - it simply says that the library failed to load.  While it's possible that the single change made to the code for this to compile is the culprit, I am somewhat skeptical of this because in a previous run I tried make the `constexpr` to `const` change in the above line on macOS and Linux as well, and the library loaded fine on those systems.  So either there is a Windows-specific issue with my workaround, or there is an unrelated failure preventing the library from loading.

I am currently trying to get this setup to work in my local Windows machine or a VM to see if I can try to see what the final `.dll` file looks like and try to narrow down why it's not loading.  Any suggestions are much appreciated!

danielnachun commented:
As an additional note, I'm not really sure if this is an issue with C++23 support - the line causing the current failure has been the same since package was created in 2020: https://github.com/cran/terra/blob/0c7eb161560cdb562580f28d0d706f3424537941/src/NA.h.  I can also see that on all 3 OSes, C++17 is the standard being used which would also rule out a C++23 issue.

Instead I'm wondering if this an issue with a very subtle API incompatibility with the MSVC C++ standard library vs libstdc++/libc++.  I haven't been able to find anything about issue so far, making it hard to pinpoint what it doesn't work.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict. Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the @ notation in a comment) if you believe this is a bug.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

danielnachun committed:
windows build with clang

danielnachun committed:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.44.9, and conda-forge-pinning 2024.12.04.17.04.39

danielnachun committed:
re-enable Windows

danielnachun committed:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.44.9, and conda-forge-pinning 2024.12.10.18.05.37

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ The recipe must have a `build/number` section.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12327781303](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12327781303). Examine the logs at this URL for more detail.</sub>

danielnachun committed:
fix recipe

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

ocefpaf committed:
Merge branch 'main' into win_build

ocefpaf committed:
refresh patch and bump

ocefpaf committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.11.13.15.40

Other tools:
- conda-build 25.11.0
- rattler-build 0.53.0
- rattler-build-conda-compat 1.4.10

danielnachun committed:
update compiler version

danielnachun committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.11.16.09.59

Other tools:
- conda-build 25.11.0
- rattler-build 0.51.0
- rattler-build-conda-compat 1.4.10

danielnachun committed:
try to remove workaround

ocefpaf committed:
Merge branch 'main' into win_build

Commit:
MNT: Re-rendered with conda-smithy 3.61.2 and conda-forge-pinning 2026.05.04.16.48.4

ocefpaf committed:
update checksum

ocefpaf committed:
remove patch
